frontend/packages/dev-console/src/components/project-access/project-access-form-utils.ts

```diff
@@ -1,24 +1,27 @@
 import * as _ from 'lodash';
-import { UserRoleBinding, UserRole } from './project-access-form-utils-types';
+import { UserRoleBinding, RoleBinding } from './project-access-form-utils-types';
 
-export const filterRoleBindings = (roleBindings, roles) => {
-  return _.filter(roleBindings.data, (user: UserRole) => _.keys(roles).includes(user.roleRef.name));
+export const filterRoleBindings = (roleBindings: RoleBinding[], roles) => {
+  return _.filter(roleBindings, (user: RoleBinding) => _.keys(roles).includes(user.roleRef.name));
 };
 
-export const getUserRoleBindings = (roleBindings) => {
+export const getUsersFromSubject = (user: RoleBinding) => {
+  const users: UserRoleBinding[] = user.subjects?.reduce((acc, obj) => {
+    acc.push({
+      roleBindingName: user.metadata.name,
+      user: obj.name,
+      role: user.roleRef.name,
+    });
+    return acc;
+  }, []);
+
+  return users;
+};
+
+export const getUserRoleBindings = (roleBindings: RoleBinding[]) => {
   let userRoleBindings: UserRoleBinding[] = [];
   roleBindings.map(
-    (user: UserRole) =>
-      (userRoleBindings = [
-        ...userRoleBindings,
-        ...[
-          {
-            roleBindingName: user.metadata.name,
-            user: user.subjects[0].name,
-            role: user.roleRef.name,
-          },
-        ],
-      ]),
+    (user: RoleBinding) => (userRoleBindings = [...userRoleBindings, ...getUsersFromSubject(user)]),
```

Comment:
Its similar to before, but the result of a map function should be used, otherwise can we switch to a .forEach here? Isn't this similar to this code?

```tsx
export const getUserRoleBindings = (roleBindings: RoleBinding[]) => {
  const userRoleBindings: UserRoleBinding[] = [];
  roleBindings.forEach(
    (user: RoleBinding) => userRoleBindings.push(...getUsersFromSubject(user)),
  );
  return userRoleBindings;
};
```

(untested code). Wdyt? :)

Reply:
used `.flat` here 
added the test for this util in `project-access-form-utils.spec.ts`